server/src/main/scala/org/broadinstitute/dsp/workbench/welder/server/ObjectService.scala

```diff
@@ -30,21 +44,104 @@ class ObjectService(googleStorageService: GoogleStorageService[IO]) extends Http
       } yield resp
   }
 
-  def localize(req: Localize): IO[Unit] = ???
+  def localize(req: Localize): IO[Unit] = {
+    val res = Stream.emits(req.entries).map {
+      entry =>
+        googleStorageService.getObject(entry.bucketNameAndObjectName.bucketName, entry.bucketNameAndObjectName.blobName, None) //get file from google
+          .through(io.file.writeAll(Paths.get(entry.localObjectPath.asString), blockingEc)) //write file to local disk
+    }.parJoin(10)
+
+    res.compile.drain
+  }
 
-  def checkMeta(req: GetMetadataRequest): IO[Unit] = ???
+  def checkMetadata(req: GetMetadataRequest): IO[MetadataResponse] =
+    for {
+      traceId <- IO(TraceId(randomUUID()))
+      storageLinks <- readJsonFileToA[IO, List[StorageLink]](pathToStorageLinks).compile.lastOrError
+      storageLink = storageLinks.find(_.localBaseDirectory == req.localObjectPath) //TODO: handle recursive flag
```

Comment:
IF we're in execution mode, it should also return the storageLink

Reply:
I need more context about this comment

Reply:
I think this would probably be a follow-up PR, but I added some examples to the design doc appendix: https://docs.google.com/document/d/1rTq3DwsB2h7l_9f2I5pcpa9Z-PJJkMBw7APBqYs1NW8/edit#heading=h.mchwz84nu1kh

After looking at the examples I'm going to propose some tweaks to this aspect of the API in the doc, curious about your thoughts on this (see comment).